core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocateEmptyPrimaryAllocationCommand.java

```diff
@@ -72,6 +74,18 @@ public String name() {
         return NAME;
     }
 
+    @Override
+    public String description() {
+        return String.format(Locale.ROOT, "allocate an empty primary for [%1$s][%2$s] on node [%3$s]", index, shardId, node);
+    }
+
+    @Override
+    public Optional<String> getMessage() {
```

Comment:
My thinking was that we didn't want to change the wire format of RerouteExplanation by adding the message to it (I think Yannick mentioned this), so here seemed like an okay place to generate the message. 

Do you mean putting this message into one of the field of the Decision that the RerouteExplanation is constructed with? E.g.

```java
public RerouteExplanation execute(RoutingAllocation allocation, boolean explain) {
    // ...
    return new RerouteExplanation(this, allocation.decision(Decision.YES, name() + " (allocation command)", "allocated an empty primary for ... rest of the message"));
}
```